### PR TITLE
Add welcome intro to panel view

### LINKED ISSUE #53: Design alveus intro for the panel view
Design the wireframe to incorporate the alveus intro into the panel view.
- Probably needs some nav bar at the bottom
- Intro probably needs to be a new page
  - Could also be a popup similar to the card pop up

### PR DESCRIPTION
**Pull request by umami-umami**
## Overview

This PR resolves #53 to add the welcome intro that is used in the overlay extension to the panel extension. A button was added to the nav bar of the panel app. Clicking the button displays the welcome card as an overlay. 

## Changes 
* Moved `Welcome` component from overlay extension components to shared components
* Modified `Welcome` social icons to allow overflow for small screens
* Added info icon as a button in the nav component 
* Created new component in the panel that re-uses the shared `Welcome` component 
* Show / hide the new welcome overlay component in the panel based on button click

## Screenshots

<details>

<summary>Panel</summary>

Screen Shot 2024-01-31 at 4 44 18 PM

</details>
<details>

<summary>Mobile (Small screen)</summary>

Screen Shot 2024-01-31 at 4 43 16 PM

</details>
<details>

<summary>Video Overlay</summary>

Screen Shot 2024-01-31 at 4 44 48 PM

</details>

### CODE REVIEW

File: src/pages/panel/components/nav/Nav.tsx [hunk 1]
```diff
@@ -2,11 +2,25 @@ import alveus from "../../../../assets/alveus.png";
...
+import IconInfo from "../../../../components/icons/IconInfo";
+
+interface NavProps {
+  onShowWelcomeCardClicked: () => void;
+}
+
+export default function Nav({ onShowWelcomeCardClicked }: NavProps) {
   return (
     <nav className={styles.nav}>
+      <div className={styles.filler} />
```
Comment (umami-umami) [inline src/pages/panel/components/nav/Nav.tsx:14]: I'm admittedly pretty new to web dev, so I'm not sure if there's a better way to make the info icon right-aligned while maintaining the center justification of the logo and nav title. LMK if there's something better I can do.

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
This is generally looking great and is what I had in mind for this! Most of my comments are nits :)

One general thing (couldn't find a logical place in the files to leave the comment), could we add a transition when the overlay is rendered on the panel, the same as the cards (they slide in)?

File: src/pages/panel/components/nav/nav.module.scss [hunk 2]
```diff
@@ -25,4 +25,14 @@
...
+
+  .filler {
+    margin-right: auto;
+  }
+
+  .info {
+    color: $primary-text;
+    margin-left: auto;
+    padding: 0.25rem 0.25rem;
+  }
```
[the remark this answers is not in the document]
Comment (MattIPv4) [reply src/pages/panel/components/nav/nav.module.scss:37]: I think with this change things might still be a bit funky just due to the sheer tiny screen size that Twitch shows the panel on -- we might need to add a media query to the nav styling to account for that:

```diff
- justify-content: center;
- width: 100vw;
- padding: 0.5rem;
+ justify-content: start;
+ right: 0;
+ left: 0;
+ padding: 0.5rem calc(2rem + 0.5rem + 20px) 0.5rem 0.5rem; // account for info button (right only)
+ 
+ @media (min-width: 25rem) {
+   justify-content: center;
+   padding: 0.5rem calc(2rem + 0.5rem + 20px); // account for info button (balanced)
+ }
```

I _think_ this should result in the logo/heading being on the left for the Twitch panel width, and then being centered once there is space if the panel ever ended up on a bigger screen

[the remark this answers is not in the document]
Comment (umami-umami) [reply src/pages/panel/components/nav/nav.module.scss:37]: Nice! Looks like that worked as you intended. In really small screen sizes, the logo/heading is on the left.

**Review by MattIPv4 — APPROVED** [review]
LGTM, tyvm

**Pull request merged**